src/Native/Unix/System.Native/pal_networkstatistics.c

```diff
@@ -482,6 +486,54 @@ int32_t SystemNative_GetNativeIPInterfaceStatistics(char* interfaceName, NativeI
             retStats->OutMulticastPackets = systemStats.ifi_omcasts;
             retStats->InDrops = systemStats.ifi_iqdrops;
             retStats->InNoProto = systemStats.ifi_noproto;
+
+            retStats->Flags = 0;
+            if (ifHdr->ifm_flags & IFF_UP)
+            {
+                retStats->Flags |= InterfaceUp;
+#if HAVE_NET_IFMEDIA_H
+                int fd =  socket(AF_INET, SOCK_DGRAM, 0);
+                if (fd < 0) {
+                    retStats->Flags |= InterfaceError;
+                }
+                else
+                {
+                    struct ifmediareq ifmr;
+                    memset(&ifmr, 0, sizeof(ifmr));
+                    strncpy(ifmr.ifm_name, interfaceName, sizeof(ifmr.ifm_name));
+
+                    if ((ioctl(fd, SIOCGIFMEDIA, (caddr_t)&ifmr) < 0) || ((ifmr.ifm_status & IFM_AVALID) == 0))
+                    {
+                        if (errno == EOPNOTSUPP)
```

Comment:
Is it possible if ioctl succeeded (but the subsequent check failed) that errno could still be EOPNOTSUPP from some previous, unrelated operation?  Or does ioctl guarantee errno will be reset to 0 or something if it succeeds?

Reply:
system call would set errno on failure but if it succeeds value of errno is not defined e.g. it could have old value as you suggested. I'll fix it.